### PR TITLE
feat: [AXIMST-111] Textbooks: textbooks index page

### PR DESCRIPTION
**Pull request by vladislavkeblysh**
- Added Textbook page
- Added textbook empty placeholder component
- Added textbook card component
- Added tests
- Added data proccessing

[AXIMST-111](https://youtrack.raccoongang.com/issue/AXIMST-111)

### CODE REVIEW

File: src/textbooks/Textbooks.jsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      </Row>
+    );
+  }
+
+  return (
+    <Container size="xl" className="px-4">
+      <section className="mb-4 mt-5">
+        <SubHeader
+          title={intl.formatMessage(messages.headingTitle)}
+          breadcrumbs={<Breadcrumb ariaLabel="Textbook breadcrumb" links={breadcrumbs} />}
```
Comment (PKulkoRaccoonGang) [inline src/textbooks/Textbooks.jsx:43]: [nit]: Let's add a translation for this aria-label

File: src/textbooks/data/api.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+/**
+ * Get textbooks for course.
+ * @param {string} courseId
+ * @returns {Promise<Object>}
+ */
+export async function getTextbooks(courseId) {
+  const { data } = await getAuthenticatedHttpClient().get(
+    getTextbooksApiUrl(courseId),
+  );
```
Comment (PKulkoRaccoonGang) [inline src/textbooks/data/api.js:17]: [optional]: Formatting similar to other pages

```suggestion
  const { data } = await getAuthenticatedHttpClient()
    .get(getTextbooksApiUrl(courseId));
```

File: src/textbooks/empty-placeholder/EmptyPlaceholder.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
```
Comment (PKulkoRaccoonGang) [inline src/textbooks/empty-placeholder/EmptyPlaceholder.jsx:1]: [nit]: Please remove this unnecessary React import

File: src/textbooks/empty-placeholder/EmptyPlaceholder.test.jsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
```
Comment (PKulkoRaccoonGang) [inline src/textbooks/empty-placeholder/EmptyPlaceholder.test.jsx:1]: [nit]: Please remove this unnecessary React import

File: src/textbooks/empty-placeholder/EmptyPlaceholder.test.jsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+    expect(getByText(messages.title.defaultMessage)).toBeInTheDocument();
+    expect(getByRole('button', { name: messages.button.defaultMessage })).toBeInTheDocument();
+  });
+
+  it('calls the onCreateNewTextbook function when the button is clicked', () => {
+    const { getByRole } = renderComponent();
+
+    const addButton = getByRole('button', { name: messages.button.defaultMessage });
+    fireEvent.click(addButton);
```
Comment (PKulkoRaccoonGang) [inline src/textbooks/empty-placeholder/EmptyPlaceholder.test.jsx:28]: [important]: We agreed to use `userEvent` in this project. 

```suggestion
    userEvent.click(addButton);
```

File: src/textbooks/hooks.jsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { useContext, useEffect } from 'react';
+import { AppContext } from '@edx/frontend-platform/react';
+import { useIntl } from '@edx/frontend-platform/i18n';
+
+import { getTextbooksData, getLoadingStatus } from './data/selectors';
+import { fetchTextbooksQuery } from './data/thunk';
+import { RequestStatus } from '../data/constants';
+import messages from './messages';
```
Comment (PKulkoRaccoonGang) [inline src/textbooks/hooks.jsx:9]:
```suggestion
import { RequestStatus } from '../data/constants';
import { getTextbooksData, getLoadingStatus } from './data/selectors';
import { fetchTextbooksQuery } from './data/thunk';
import messages from './messages';
```

File: src/textbooks/textbook-card/TextbooksCard.jsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
+import React from 'react';
```
Comment (PKulkoRaccoonGang) [inline src/textbooks/textbook-card/TextbooksCard.jsx:1]: [nit]: Please remove this unnecessary React import

File: src/textbooks/textbook-card/TextbooksCard.test.jsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
```
Comment (PKulkoRaccoonGang) [inline src/textbooks/textbook-card/TextbooksCard.test.jsx:1]: [nit]: Please remove this unnecessary React import

**Review by PKulkoRaccoonGang — CHANGES_REQUESTED** [review]

File: src/textbooks/textbook-card/TextbooksCard.test.jsx (hunk 8 above)
Comment (vladislavkeblysh) [reply src/textbooks/textbook-card/TextbooksCard.test.jsx:1]: Fixed

File: src/textbooks/textbook-card/TextbooksCard.jsx (hunk 7 above)
Comment (vladislavkeblysh) [reply src/textbooks/textbook-card/TextbooksCard.jsx]: Fixed

File: src/textbooks/hooks.jsx (hunk 6 above)
Comment (vladislavkeblysh) [reply src/textbooks/hooks.jsx]: Fixed

File: src/textbooks/empty-placeholder/EmptyPlaceholder.test.jsx (hunk 5 above)
Comment (vladislavkeblysh) [reply src/textbooks/empty-placeholder/EmptyPlaceholder.test.jsx]: Fixed

File: src/textbooks/empty-placeholder/EmptyPlaceholder.test.jsx (hunk 4 above)
Comment (vladislavkeblysh) [reply src/textbooks/empty-placeholder/EmptyPlaceholder.test.jsx:1]: Fixed

File: src/textbooks/empty-placeholder/EmptyPlaceholder.jsx (hunk 3 above)
Comment (vladislavkeblysh) [reply src/textbooks/empty-placeholder/EmptyPlaceholder.jsx]: Fixed

File: src/textbooks/data/api.js (hunk 2 above)
Comment (vladislavkeblysh) [reply src/textbooks/data/api.js]: Fixed

File: src/textbooks/Textbooks.jsx (hunk 1 above)
Comment (vladislavkeblysh) [reply src/textbooks/Textbooks.jsx]: Fixed

File: src/textbooks/Textbooks.jsx [hunk 9]
```diff
@@ -0,0 +1,92 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (vladislavkeblysh) [reply src/textbooks/Textbooks.jsx]: Fixed

File: src/textbooks/Textbook.test.jsx [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+import { AppProvider } from '@edx/frontend-platform/react';
+import { initializeMockApp } from '@edx/frontend-platform';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+
+import initializeStore from '../store';
+import { executeThunk } from '../utils';
+import { getTextbooksApiUrl } from './data/api';
+import { fetchTextbooksQuery } from './data/thunk';
+import { textbooksMock } from './__mocks__';
+import { Textbooks } from './index';
```
[the remark this answers is not in the document]
Comment (vladislavkeblysh) [reply src/textbooks/Textbook.test.jsx]: Fixed

File: .gitignore [hunk 11]
```diff
@@ -20,3 +20,4 @@ temp/babel-plugin-react-intl
 /temp
 /.vscode
 /module.config.js
+/module.dev-tutor.config.js
```
[the remark this answers is not in the document]
Comment (vladislavkeblysh) [reply .gitignore:23]: Fixed

File: src/textbooks/textbook-card/TextbooksCard.jsx [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+  DeleteOutline as DeleteIcon,
+} from '@edx/paragon/icons';
+
+import messages from './messages';
+
+const TextbookCard = ({ title, chapters }) => {
+  const intl = useIntl();
+
+  return (
+    <div className="textbook-card" data-testid="textbook-card">
```
Comment (monteri) [inline src/textbooks/textbook-card/TextbooksCard.jsx:20]: Can we use here `Card` component from Paragon. The idea is to reuse Paragon components where it is possible.
I would consider to use the `Card` wrapper and the `Card.Header` with the `ActionRow` here.

For reference:

Similar component:
https://github.com/raccoongang/frontend-app-course-authoring/blob/4f2cf7308817488ea7da464f7d85a4cba895c586/src/course-unit/course-xblock/CourseXBlock.jsx#L39

Design for XBlock:
https://www.figma.com/file/YeKFwSpyLaJFDs3f3p3TSa/Studio-1%3A1-mock-ups?node-id=599%3A23626&mode=dev

Comment (vladislavkeblysh) [reply src/textbooks/textbook-card/TextbooksCard.jsx]: Fixed

**Review by monteri — APPROVED** [review]

**Pull request merged**